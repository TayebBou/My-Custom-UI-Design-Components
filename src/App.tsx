import { FC } from 'react';
import styles from './App.module.css';
import BigImageSlider from './components/molecules/ImageSlider/BigImageSlider';
import data from './server/data.json'
import { IPhoto } from './shared/types/photo.model';

const App:FC = () => {
  return (
    <div className={styles.app}>
      <BigImageSlider photos={data.photos as IPhoto[]} />
    </div>
  );
}

export default App;
