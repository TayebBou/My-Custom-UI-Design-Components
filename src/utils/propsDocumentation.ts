export const buttonPropsDocumentation = [
  {
    name: "children",
    type: "React.ReactNode",
    description: "The content inside the button, usually text or an icon.",
    inputType: "none",
  },
  {
    name: "onClick",
    type: "() => void",
    description: "The function to handle button click events. Optional.",
    inputType: "none",
  },
  {
    name: "className",
    type: "string",
    description: "Additional CSS class for styling the button. Optional.",
    inputType: "none",
  },
  {
    name: "type",
    type: "ButtonTypes (BUTTON | RESET | SUBMIT)",
    description: 'Defines the button type. Defaults to "button".',
    inputType: "none",
  },
  {
    name: "disabled",
    type: "boolean",
    description: "Disables the button if true. Defaults to false.",
    inputType: "checkbox",
  },
];

export const iconPropsDocumentation = [
  {
    name: "src",
    type: "string",
    description: "The source URL of the icon image.",
    inputType: "none",
  },
  {
    name: "alt",
    type: "string",
    description:
      "The alt text for the icon, providing accessibility information.",
    inputType: "none",
  },
  {
    name: "className",
    type: "string",
    description: "Optional CSS class for additional styling of the icon.",
    inputType: "none",
  },
  {
    name: "size",
    type: "string",
    description:
      "Defines the size of the icon. Can be any valid CSS size value.",
    inputType: "select",
    options: ["16px", "24px", "32px", "48px"],
    defaultValue: "32px",
  },
];

export const navBarPropsDocumentation = [
  {
    name: "children",
    type: "React.ReactNode",
    description:
      "Le contenu à afficher à l'intérieur de la barre de navigation.",
    inputType: "none",
  },
  {
    name: "fixed",
    type: "boolean",
    description:
      "Définit si la barre de navigation doit être fixée en haut de la page. Valeur par défaut: false.",
    inputType: "none",
  },
  {
    name: "className",
    type: "string",
    description:
      "Classe CSS supplémentaire pour personnaliser le style de la barre de navigation. Optionnel.",
    inputType: "none",
  },
];

export const fullScreenPropsDocumentation = [
  {
    name: "element",
    type: "HTMLElement",
    description:
      "L'élément qui doit être affiché en plein écran. Si non fourni, l'élément principal du document sera utilisé.",
    inputType: "none",
  },
  {
    name: "className",
    type: "string",
    description:
      "Classe CSS supplémentaire pour personnaliser le style du bouton plein écran. Optionnel.",
    inputType: "none",
  },
];

export const logoPropsDocumentation = [
  {
    name: "className",
    type: "string",
    description:
      "Classe CSS supplémentaire pour personnaliser le style du logo. Optionnel.",
    inputType: "none",
  },
];

export const bigImageSliderPropsDocumentation = [
  {
    name: "photos",
    type: "IPhoto[]",
    description:
      "An array of photo objects. Each object must contain a 'photos' array, and each photo inside it should have properties like 'url_full'.",
    inputType: "none",
  },
];

export const modalPropsDocumentation = [
  {
    name: "onDisplay",
    type: "boolean",
    description:
      "Controls whether the modal is displayed. Should be `true` to show the modal and `false` to hide it.",
    inputType: "checkbox",
  },
  {
    name: "onClose",
    type: "() => void",
    description:
      "Function to handle closing the modal. Called after the close button is clicked or the background is clicked.",
    inputType: "none",
  },
  {
    name: "children",
    type: "React.ReactNode",
    description:
      "The content inside the modal. Can include any valid React elements.",
    inputType: "none",
  },
  {
    name: "className",
    type: "string",
    description:
      "Optional additional CSS classes for styling the modal. Empty by default.",
    inputType: "none",
  },
  {
    name: "title",
    type: "string",
    description:
      "Optional title for the modal. Displays at the top in the header section. Empty by default.",
    inputType: "text",
    defaultValue: "Sample modal",
  },
];

export const sideBarPropsDocumentation = [
  {
    name: "children",
    type: "React.ReactNode",
    description: "The content to be displayed inside the sidebar.",
    inputType: "none",
  },
  {
    name: "onClose",
    type: "() => void",
    description:
      "Function to be executed when the sidebar is closed, typically to hide the sidebar.",
    inputType: "none",
  },
  {
    name: "closeSideBar",
    type: "boolean",
    description:
      "A flag that triggers the sidebar to close. When true, the sidebar starts its closing animation and executes the onClose function.",
    inputType: "none",
  },
];

export const imageSliderPropsDocumentation = [
  {
    name: "photos",
    type: "IPhoto[]",
    description:
      "An array of photo objects. Each object must contain a 'photos' array, and each photo inside it should have properties like 'url_full'.",
    inputType: "none",
  },
];
