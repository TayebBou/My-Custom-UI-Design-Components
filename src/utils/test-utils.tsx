import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import { Store } from '@reduxjs/toolkit'
import type { PreloadedState } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import globalStore from '../config/store'
import { IRootState } from '../shared/types/rootState.model'
// As a basic setup, import your same slice reducers
// import userReducer from '../features/users/userSlice'

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<IRootState>
  store?: Store
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {
      navBar: {
        isSmallScreen: false,
        displaySide: false,
        closeSideBar: false,
        isNavMenuExpanded: false,
        isNavMenuExiting: false,
      },
    },
    // Automatically create a store instance if no store was passed in
    store = globalStore,
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}
