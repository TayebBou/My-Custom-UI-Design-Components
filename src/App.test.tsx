import { renderWithProviders } from "./utils/test-utils";
import App from "./App";

describe("App component", () => {
  test("should render correctly", () => {
    // Arrange
    const { container } = renderWithProviders(<App />);
    // Assert
    expect(container).toMatchSnapshot();
  });
});
