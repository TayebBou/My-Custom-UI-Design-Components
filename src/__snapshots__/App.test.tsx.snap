// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div>
  <div
    class="navbar navbar"
  >
    <a
      class="logo logo"
      href="/"
    >
      <img
        alt="logo"
        class=""
        src="ui.png"
        style="width: 44px; height: 44px; filter: none;"
      />
      <h1
        class="h1"
      >
        My Custom 
        <br />
        <span
          class="span"
        >
          UI Design Components
        </span>
      </h1>
    </a>
    <div
      class="parent-div"
    >
      <button
        class="button menu dropdownmenu"
        type="button"
      >
        <img
          alt="collapse"
          class=""
          src="next.png"
          style="width: 32px; height: 32px; filter: invert(1);"
        />
        <h1
          class="h1"
        >
          UI Organisms
        </h1>
      </button>
    </div>
    <div
      class="parent-div"
    >
      <button
        class="button menu dropdownmenu"
        type="button"
      >
        <img
          alt="collapse"
          class=""
          src="next.png"
          style="width: 32px; height: 32px; filter: invert(1);"
        />
        <h1
          class="h1"
        >
          UI Molecules
        </h1>
      </button>
    </div>
    <div
      class="parent-div"
    >
      <button
        class="button menu dropdownmenu"
        type="button"
      >
        <img
          alt="collapse"
          class=""
          src="next.png"
          style="width: 32px; height: 32px; filter: invert(1);"
        />
        <h1
          class="h1"
        >
          UI Atoms
        </h1>
      </button>
    </div>
  </div>
  <div
    class="app"
  >
    <h2>
      ImageSlider Component
    </h2>
    <div
      class="wrapper"
    >
      <div
        class="fullscreen"
      >
        <div
          class="full-images"
          data-testid="bigImagesDiv"
        >
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_1.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_2.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_3.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_4.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_5.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_6.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_7.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_8.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_9.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_10.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <img
            alt="car outside"
            src="/images/MERCEDES_CLASSE_C_11.webp"
            style="max-width: 100%; transition: 0.5s; transform: translateX(0%);"
          />
          <button
            class="button slider-button right"
            type="button"
          >
            &gt;
          </button>
          <button
            class="button slider-button left"
            disabled=""
            type="button"
          >
            &lt;
          </button>
          <span
            class="span"
          >
            1
             / 
            11
          </span>
          <button
            class="button button "
            type="button"
          >
            <img
              alt="full screen"
              class="img"
              src="full_screen.png"
              style="filter: none;"
            />
          </button>
        </div>
        <div
          class="mini-images"
          data-testid="imagesDiv"
        >
          <button
            class="button slider-button"
            disabled=""
            type="button"
          >
            &lt;
          </button>
          <div
            class="images"
            data-testid="miniImagesDiv"
          >
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_1.webp"
              style="max-width: 92px; margin: 1.5em 0px; outline: solid 0.2em white; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_2.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_3.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_4.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_5.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_6.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_7.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_8.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_9.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_10.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
            <img
              alt="car outside vignette"
              src="/images/MERCEDES_CLASSE_C_11.webp"
              style="max-width: 92px; margin: 1.5em 0px; transition: transform 0.5s; transform: translateX(0px);"
            />
          </div>
          <button
            class="button slider-button"
            type="button"
          >
            &gt;
          </button>
        </div>
      </div>
    </div>
    <div>
      <h3>
        Props Documentation
      </h3>
      <table
        class="table"
      >
        <thead>
          <tr>
            <th
              class="header"
            >
              Prop Name
            </th>
            <th
              class="header"
            >
              Type
            </th>
            <th
              class="header"
            >
              Description
            </th>
            <th
              class="header"
            >
              Modify
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="cell"
            >
              photos
            </td>
            <td
              class="cell"
            >
              IPhoto[]
            </td>
            <td
              class="cell"
            >
              An array of photo objects. Each object must contain a 'photos' array, and each photo inside it should have properties like 'url_full'.
            </td>
            <td
              class="cell"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
